### ISSUE TITLE
Separate subpackages into their own repositories and make payum/payum composer.json only repository

### ISSUE THREAD

**Issue by Tetragramat** [Payum/Payum#912]
Installing `payum/payum` should be discuragued in favor of installing `payum/core` and gateways separately.
Having `composer.json` in `payum/payum` to be merge of requirements of all it's packages is error prone and so sub-packages should be separated into their own repositories and `payum/payum` should turn into `composer.json` repository that requires it's original subpackages as backward compatibility.

**Comment by pierredup** [comment]
> Installing payum/payum should be discuragued in favor of installing payum/core and gateways separately.

I agree with this and is probably something that can just be updated in the documentation, and the package marked as abandoned in Packagist (with `payum/core` suggested as a replacement).

> sub-packages should be separated into their own repositories and payum/payum should turn into composer.json repository that requires it's original subpackages

I think keeping `payum/payum` as a mono-repo that contains all the packages has its benefits and it makes a lot of integration and development a lot easier. This repo is then split into the separate sub-repositories, which can be installed on its own

> every release of new version creates new version of all gateway packages even thought there was no change in their code

This can be addressed in the subtree-split process when creating a new tag so that only sub-packages that have changes gets a new tag. This is something that Symfony also did recently. Previously, any new version in `symfony/symfony` created a new version in all the components, until recently where they started to only create new versions of components if there was changes. I think we can follow the same rule here.

**Comment by makasim** [comment]
> I think keeping payum/payum as a mono-repo that contains all the packages has its benefits and it makes a lot of integration and development a lot easier. This repo is then split into the separate sub-repositories, which can be installed on their own

True. Ease of development is main reason this repo exists.

> every release of new version creates new version of all gateway packages even thought there was no change in their code

That's not exactly true. The new tag is created only if there are changes in the component since last tag. See https://github.com/Payum/Payum/blob/master/bin/release#L46
